### DISCUSSION TITLE
version-vs-versionsuffix-vs-packageversion-what-do-they-all-mean/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#32] [Announcements]
# version-vs-versionsuffix-vs-packageversion-what-do-they-all-mean/

In this post I look at the various version numbers you can set when building a .NET Core project, such as Version, VersionSuffix, and PackageVersion.

https://andrewlock.net/version-vs-versionsuffix-vs-packageversion-what-do-they-all-mean/

**Comment by greijner** [comment]
Great blog! As well as many others, very informational, thank you!

However, regarding this blogpost, I see that the `FileVersion` defaults to the `Version` instead of the `AssemblyVersion`. The `FileVersion` is set this way after running `dotnet publish`, without having the service being published as a single executable file.

Regards,
Gijs

**Reply by greijner** [reply]
I'm sorry, I was mistaken. It's the `ProductVersion` that defaults to the `Version`. Your blogpost about the `FileVersion` is correct.

**Comment by stavnskov** [comment]
It seems like AssemblyInformationalVersion cannot be set through dotnet build command line to e.g. 1.2.3-comment+sha.abc123, but only through a AssemblyInfo.cs file

**Comment by yodasad** [comment]
Hello! Thank you for the great article!

Could you please update the part about InformationalVersion with the following:

If the $(SourceRevisionId) property is present, it's appended to InformationalVersion. You can disable this behavior using IncludeSourceRevisionInInformationalVersion.

It is from the official documentation:
https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#generateassemblyinfo

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 09 2018, 09:34, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/32#discussioncomment-6045924)</em>

---

<p>That's odd - it works on my machine!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 12 2018, 07:35, in reply to [Georgi Parlakov](https://disqus.com/by/georgiparlakov/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/32#discussioncomment-6045934)</em>

---

<p>That's strange, I'm using the same browser and it all looks correct :/ Not sure what could have caused that...</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 15 2019, 10:15, in reply to [Brian Baker](https://disqus.com/by/brian_baker/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/32#discussioncomment-6045946)</em>

---

<p>In general, I think continuing the same AssemblyVersion approach makes sense, especially as libraries may continue to be consumed by .NET Framework apps. That said, if you're sure the app will only be used in .NET Core, you'll be fine to not worry about it :)</p>

[a comment by an automated account is not shown]

**Comment by urbasus** [comment]
According to https://learn.microsoft.com/en-us/nuget/concepts/package-versioning the (SemVer v2.0.0) version should be like major.minor.patch[-prerelease][.build], whereas you've put the prerelease and build the opposite order in this post.

**Reply by andrewlock** [reply]
The post is correct 🙂  The .NET versioning scheme predates sem-ver by about 10 years 😅 the [.build] part of the 4-part version is often left at 0 though. NuGet has supported [.build] in the prerelease segment [since approximately 2017](https://github.com/NuGet/Home/issues/1359#issuecomment-328519398). Hope that makes sense!

**Comment by Gladskih** [comment]
In addition to these many versions of versions it would be nice to know what actually the gain? For example we build are creating an containerized service. Let assume the API of the service requires to include a version of the service into a response. Is there a reason to mess with those legacy mechanisms (and which one)? Or is it better to ignore them and keep defaults and just provide an environment variable with version? By the way does it make sense to keep all separate App version, Image version, Helm chart version, Helm chart appVersion, service version? It feels like only reusable artifact needs to be identified by a version. Thus here should be an image version, and release version (in terms of 12 factors app where "release = build+config"). Actually I would extend it further to a version of service or version of installation kind of "release + particular env", because there are no two identical environments.
